pkg/virt-launcher/virtwrap/network/cloud-init-network_test.go

```diff
@@ -0,0 +1,225 @@
+// Inspired by pkg/virt-launcher/virtwrap/network/podinterface_test.go
+
+package network
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net"
+	"os"
+	"strconv"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/vishvananda/netlink"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"gopkg.in/yaml.v2"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/log"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+)
+
+var _ = Describe("cloud-init network", func() {
+	var mockNetwork *MockNetworkHandler
+	var ctrl *gomock.Controller
+
+	var tmpDir string
+
+	log.Log.SetIOWriter(GinkgoWriter)
+
+	BeforeEach(func() {
+		tmpDir, _ := ioutil.TempDir("", "cloudinittest")
+		setInterfaceCacheFile(tmpDir + "/cache-%s.json")
```

Comment:
I'm curious, which part of ```GenNetworkFile``` touches disk. It looks to me that the function is quite pure in that it receives a VMI spec and returns a slice of bytes, but it doesn't seem to involve any disk state changes. Am I missing something?

Reply:
I tapped into the cloud-init iso generation that already existed in manager.go in preStartHook.